packages/framer-motion/src/render/html/utils/build-transform.ts

```diff
@@ -31,20 +37,12 @@ export function buildTransform(
     // Transform keys into their default order - this will determine the output order.
     transformKeys.sort(sortTransformProps)
 
-    // Track whether the defined transform has a defined z so we don't add a
-    // second to enable hardware acceleration
-    let transformHasZ = false
-
     // Loop over each transform and build them into transformString
-    const numTransformKeys = transformKeys.length
-    for (let i = 0; i < numTransformKeys; i++) {
-        const key = transformKeys[i]
+    for (const key of transformKeys) {
         transformString += `${translateAlias[key] || key}(${transform[key]}) `
-
-        if (key === "z") transformHasZ = true
     }
 
-    if (!transformHasZ && enableHardwareAcceleration) {
+    if (enableHardwareAcceleration && !transform.z) {
```

Comment:
Should this be `transform.z !== undefined`? It can be 0 right 🤔

Reply:
Ah yeah but in this case if it were `0` it wouldn't have been included in the list and this conditional is adding `translateZ(0)` anyway

Reply:
But in case it was 0, we would be adding double `translateZ(0)`, right? (on line:42 `transformString += `${translateAlias[key] || key}(${transform[key]})`)

Reply:
Yeah exactly - if its `0` it's already been added and 

```
    if (enableHardwareAcceleration && !transform.z) {
        transformString += "translateZ(0)"
    }
```

Doesn't actually run (because `!transform.z`), so we don't add a double `translateZ(0)`